src/store/audio.ts

```diff
@@ -188,69 +189,35 @@ export function getCharacterInfo(
 }
 
 /**
- * configを参照して割り当てるべきpresetKeyとそのPresetを適用すべきかどうかを返す
+ * 与えたAudioItemを元に、Presetを適用した新しいAudioItemを返す
  */
-export function determineNextPresetKey(
-  state: State,
-  voice: Voice,
-  presetKeyCandidate: PresetKey | undefined,
-  shouldCopyBaseAudioItem: boolean,
-  isVoiceChanged = false
-): {
-  nextPresetKey: PresetKey | undefined;
-  shouldApplyPreset: boolean;
-} {
-  const defaultPresetKeyForCurrentVoice =
-    state.defaultPresetKeys[VoiceId(voice)];
-
-  const isDefaultPreset = Object.values(state.defaultPresetKeys).some(
-    (key) => key === presetKeyCandidate
-  );
-
-  // コピーすべきBaseAudioItemがない＝初回作成時
-  if (!shouldCopyBaseAudioItem) {
-    return {
-      nextPresetKey: defaultPresetKeyForCurrentVoice,
-      shouldApplyPreset: state.experimentalSetting.enablePreset,
-    };
-  }
-
-  // ボイス切り替え時
-  if (isVoiceChanged) {
-    if (state.experimentalSetting.shouldApplyDefaultPresetOnVoiceChanged) {
-      // デフォルトプリセットを適用する
-      return {
-        nextPresetKey: defaultPresetKeyForCurrentVoice,
-        shouldApplyPreset: true,
-      };
-    }
-
-    // 引き継ぎ元が他スタイルのデフォルトプリセットだった場合
-    // 別キャラのデフォルトプリセットを引き継がないようにする
-    // それ以外は指定そのまま
-    return {
-      nextPresetKey: isDefaultPreset
-        ? defaultPresetKeyForCurrentVoice
-        : presetKeyCandidate,
-      shouldApplyPreset: false,
-    };
-  }
-
-  // 以下はAudioItemコピー時
-
-  if (state.inheritAudioInfo) {
-    // パラメータ引継ぎがONならそのまま引き継ぐ
-    return {
-      nextPresetKey: presetKeyCandidate,
-      shouldApplyPreset: false,
-    };
-  }
-
-  // それ以外はデフォルトプリセットを割り当て、適用するかはプリセットのON/OFFに依存
-  return {
-    nextPresetKey: defaultPresetKeyForCurrentVoice,
-    shouldApplyPreset: state.experimentalSetting.enablePreset,
-  };
+export function applyAudioPresetToAudioItem(
+  audioItem: AudioItem | undefined,
+  presetItem: Preset | undefined
+) {
+  if (
+    audioItem == undefined ||
+    audioItem.presetKey == undefined ||
+    audioItem.query == undefined
+  )
+    return;
+  if (presetItem == undefined) return;
+
+  const newAudioItem = { ...audioItem };
```

Comment:
deepcopyにならないのでqueryの参照がおなじになっちゃうかもです。
他の箇所ではjson.stringfyで複製したりしています。（パフォーマンス的に微妙そうですが。。）

```suggestion
  const newAudioItem = JSON.parse(JSON.stringify(audioItem)) as AudioItem;
```

Reply:
https://github.com/VOICEVOX/voicevox/blob/fad2e12adb8658c59fe37ceaf87b218a3aca84b6/src/store/audio.ts#L217

ここは後ろの方で上書きしているので問題なさそうです！が、一発でdeepCopyしちゃった方が誤解がなくてよさそうですね。
テキスト複製で `deep-copy` 入れたのでそれでdeepCopyするようにしちゃいますね

Reply:
（コメント箇所が別だったので補足）
型の問題でやっぱりshallow copyするが、定義箇所は近づけておく

```
  const newAudioItem = { ...audioItem };
  newAudioItem.query = { ...audioItem.query, ...audioInfos };
```
こういう並びになる